lib/core/Stage.hpp

```diff
@@ -51,6 +52,21 @@ class Stage : public kotekanLogging {
      */
     virtual std::string dot_string(const std::string& prefix) const;
 
+    /**
+     * @brief Add newly created stage tid to thread_list for cpu usage tracking.
+     */
+    void register_tid(pid_t ptr);
+
+    /**
+     * @brief Remove a tid from thread_list.
+     */
+    void unregister_tid(pid_t ptr);
+
+    /**
+     * @brief Get tids from the current stage.
+     */
+    std::vector<pid_t> get_tids();
```

Comment:
I think we should return this as a const reference to save on the copy constructor time.  Since normally we don't need to do any modifications to the array once it's returned other than read it.